### ISSUE TITLE
readone after stateone works in different way in 2.5.0

### ISSUE THREAD

**Issue by pjb671** [sardana-org/sardana#967]
My first issue submitted here. At one beamline at MAX IV we have upgraded to sardana 2.5.0.  It seems as though the behaviour of the experimental channels has changed - a piece of hardware that returns a 1d array gets read during the acquisition, which results in incomplete data being returned. To confirm I made a super simple 1d controller with the following methods:

    def StartOne(self, axis, position=None):
        self.state = State.Moving
        print("----------------------------------STARTED -> MOVING")

    def StateOne(self,ind):  
        self.status = "test"
        #self.state = State.On   SO IT IS ALWAYS MOVING AFTER DOING STARTONE
        tup = (self.state,self.status)
        print("----------------------------------IN STATEONE", ind, self.state)
        return tup

    def ReadOne(self,ind):
        if self.state == PyTango.DevState.MOVING:
            print("------------------- STATE IS MOVING : SHOULD NEVER CALL READONE!")
            return np.zeros(MAX_BUF_SIZE)
        else:
            return np.zeros(MAX_BUF_SIZE)

In 2.3.1, it works, or rather fails, as expected. The state is always moving so a step scan hangs waiting forever, ReadOne is never called:

----------------------------------STARTED -> MOVING
----------------------------------IN STATEONE 2 6
----------------------------------IN STATEONE 2 6
...this is printed forever

In 2.4.0 and 2.5.0, the readone is called while we are still in moving state:

----------------------------------STARTED -> MOVING
----------------------------------IN STATEONE 2 6
------------------- STATE IS MOVING : SHOULD NEVER CALL READONE!
----------------------------------IN STATEONE 2 6
----------------------------------IN STATEONE 2 6
...and readOne is repeatedly called every second or so.

Is this an expected new behaviour? It fundamentally changes how we are used to things working.  I would not expect Sardana to try and read the hardware while stateone returns moving.

**Comment by reszelaz** [comment]
It looks like a bug. The C/T channels are read while acquiring to report count updates. But the 1D/2D should not be. I will check carefully and let you know.

**Comment by reszelaz** [comment]
I confirm it's a bug. Next week I will provide a fix.

**Comment by reszelaz** [comment]
You mean to not read the channel selected as Timer in the measurement group configuration?

Just loud thinking... Maybe a long `uct` execution could see how much time elapsed from the start? Of course if the controller knows how to report this time.

**Comment by dschick** [comment]
@reszelaz I just noticed that the ReadOne method is also called when the State is MOVING for a CounterTimerCtrl in timer mode.
As I understood the concept of the timer: the timer defines the start and end of the acquistion and this should be done only by the StateOne method?

**Comment by reszelaz** [comment]
> I just noticed that the ReadOne method is also called when the State is MOVING for a CounterTimerCtrl in timer mode.

Yes, for the moment, the behavior changes only based on the channel's type. So, counter/timer are read while acquiring, and 1D and 2D are not. Technically it is possible to change the behavior for counter/timers also based on the configuration to, as you say, do not read the counter/timer channels if they are configured and used as timers. I would treat it is as a separate issues if there is an interest in that.

**Comment by reszelaz** [comment]
@pjb671 I just realized that we reintroduced this bug in 2.7.0. Hope that you have not yet upgraded...

I already fixed this in SEP2: 30374266 and plan to fix it in develop as well. I don't know if hotfix release is needed...
I think you are the only one relying on this behavior so I would base the decision on your preferences.

We could just cherry-pick the commit to the develop branch and if you plan to upgrade to 2.7.0 you could use the develop branch at a stage just after the cherry-pick. After the 2.7.0 release there was nothing yet merged into the develop branch.

If you don't find it satisfactory we of course can make a hotfix release.

And if anyway you were not planning to upgrade now, and for example wait till Jul19 milestone is completed than the simple cherry-pick to develop is also enough.

Please let us know your prereferences.

Thanks and sorry for inconveniences!

**Comment by teresanunez** [comment]
Hi @reszelaz, thanks for advertising about this here.  We also had problems at DESY due to this bug ... Since we usually install the develop version and not any release it should not be a problem for us if you don't make any hotfix release, but in any case I will try to keep it in mind in case some installation 
of the Release is done at DESY.

**Comment by reszelaz** [comment]
> We also had problems at DESY due to this bug ... 

Ohh, I forgot about it. Sorry!
Since you install from develop then the simple cherry-pick would be enough for you.

Let's wait if there are other preferences and then I will make a PR with this fix.
